base/applications/rapps/installed.cpp

```diff
@@ -2,123 +2,122 @@
  * PROJECT:     ReactOS Applications Manager
  * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
  * PURPOSE:     Classes for working with installed applications
- * COPYRIGHT:   Copyright 2009 Dmitry Chapyshev         (dmitry@reactos.org)
- *              Copyright 2017 Alexander Shaposhnikov   (sanchaez@reactos.org)
- *              Copyright 2020 He Yang                  (1160386205@qq.com)
+ * COPYRIGHT:   Copyright 2009 Dmitry Chapyshev <dmitry@reactos.org>
+ *              Copyright 2017 Alexander Shaposhnikov <sanchaez@reactos.org>
+ *              Copyright 2020 He Yang <1160386205@qq.com>
+ *              Copyright 2021 Mark Jansen <mark.jansen@reactos.org>
  */
-#include "rapps.h"
 
+#include "rapps.h"
 #include "installed.h"
-
 #include "misc.h"
 
-CInstalledApplicationInfo::CInstalledApplicationInfo(BOOL bIsUserKey, REGSAM RegWowKey, HKEY hKey)
-    : IsUserKey(bIsUserKey), WowKey(RegWowKey), hSubKey(hKey)
+CInstalledApplicationInfo::CInstalledApplicationInfo(BOOL bIsUserKey, REGSAM RegWowKey, HKEY hKey, const CStringW& szKeyName)
+    : m_IsUserKey(bIsUserKey)
+    , m_WowKey(RegWowKey)
+    , m_hSubKey(hKey)
+    , m_szKeyName(szKeyName)
 {
-    // if Initialize failed, hSubKey will be closed automatically and set to zero
-
-    DWORD dwSize = MAX_PATH, dwType, dwValue;
-    BOOL bIsSystemComponent;
-    ATL::CStringW szParentKeyName;
-
-    dwType = REG_DWORD;
-    dwSize = sizeof(DWORD);
+    DWORD dwSize = 0;
+    bIsUpdate = (RegQueryValueExW(m_hSubKey, L"ParentKeyName", NULL, NULL, NULL, &dwSize) == ERROR_SUCCESS);
+}
 
-    if (RegQueryValueExW(hSubKey,
-        L"SystemComponent",
-        NULL,
-        &dwType,
-        (LPBYTE)&dwValue,
-        &dwSize) == ERROR_SUCCESS)
+CInstalledApplicationInfo::~CInstalledApplicationInfo()
+{
+    if (m_hSubKey)
     {
-        bIsSystemComponent = (dwValue == 0x1);
+        CloseHandle(m_hSubKey);
+        m_hSubKey = NULL;
     }
-    else
+}
+
+void CInstalledApplicationInfo::EnsureDetailsLoaded()
+{
+    // Key not closed, so we have not loaded details yet
+    if (m_hSubKey)
```

Comment:
You could return early and save one indentation level by `if (!m_hSubKey) return;`

Reply:
And what problem would this solve exactly?
It's not forbidden to have indentation....

Reply:
It will save a little of CPU time by shorting execution path. And its easier to maintain readability when there are less indentation level :)

Reply:
It's just about readability. In this case the CPU time wouldn't certainly be modified: both assemble to a jump instruction.

Reply:
> It will save a little of CPU time by shorting execution path. And its easier to maintain readability when there are less indentation level :)

I very much doubt that this has any relevance for the execution path.